assets/js/modules/analytics/datastore/properties.js

```diff
@@ -247,6 +248,34 @@ const baseActions = {
 			type: SET_PRIMARY_PROPERTY_TYPE,
 		};
 	},
+
+	/**
+	 * Matches a property by URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Array.<Object>}        properties Array of properties.
+	 * @param {Array.<string>|string} url        A list of URLs or a signle URL to match properties.
+	 * @return {Object|null} A property object if found. NULL if nothing is found.
+	 */
+	*matchPropertyByURL( properties, url ) {
```

Comment:
@aaemnnosttv Fair point. I don't have a strong opinion on this - we can make this a utility function if preferred.

Reply:
Let's do it in a separate ticket then? Since we already have similar functionality in the analytics-4 module.

Reply:
> This doesn't need to be a generator

It needs to be a generator because otherwise, we would need to return a regular redux action object.

Reply:
> > This doesn't need to be a generator
> 
> It needs to be a generator because otherwise, we would need to return a regular redux action object.

Oops, yeah you're right, although I don't think we use actions in this way anywhere else except for those that yield controlled actions since these must be a generator. I suppose it's okay for now then, but this as a generator only for this reason when it doesn't yield anything is an odd detail that someone else might see and "fix" which breaks it. For this reason I'd prefer we extract it sooner than later. If you feel that this would take too much time right now then let's clearly document this as an intentional requirement of the function. 

> we already have similar functionality in the analytics-4 module.

The use in `analytics-4` looks okay to me since it's actually yielding actions although maybe that could be cleaned up with a utility in the future.

Reply:
@aaemnnosttv sounds good to me. Updated.